docs/source/java/cdata.rst

```diff
@@ -220,4 +223,259 @@ Let's create a Java class to test our bridge:
+                Data.exportVector(allocator, populateFieldVectorToExport(), null, arrow_array, arrow_schema);
+            }
+        }
+
+        public static FieldVector populateFieldVectorToExport(){
+            IntVector intVector = new IntVector("int-to-export", allocator);
+            intVector.allocateNew(3);
+            intVector.setSafe(0, 1);
+            intVector.setSafe(1, 2);
+            intVector.setSafe(2, 3);
+            intVector.setValueCount(3);
+            System.out.println("[Java - side]: FieldVector: \n" + intVector);
+            return intVector;
+        }
+
+        public static void fillvectorschemarootfromjavatocpp(long schema_id, long array_id){
+            try (ArrowArray arrow_array = ArrowArray.wrap(array_id);
+                 ArrowSchema arrow_schema = ArrowSchema.wrap(schema_id) ) {
+                Data.exportVectorSchemaRoot(allocator, populateVectorSchemaRootToExport(), null, arrow_array, arrow_schema);
+            }
+        }
+
+        public static VectorSchemaRoot populateVectorSchemaRootToExport(){
+            IntVector intVector = new IntVector("age-to-export", allocator);
+            intVector.setSafe(0, 10);
+            intVector.setSafe(1, 20);
+            intVector.setSafe(2, 30);
+            VectorSchemaRoot root = new VectorSchemaRoot(Arrays.asList(intVector));
+            root.setRowCount(3);
+            System.out.println("[Java - side] VectorSchemaRoot: \n" + root.contentToTSVString());
+            return root;
+        }
+
+        public static void main(String[] args) {
+            populateFieldVectorToExport();
+            populateVectorSchemaRootToExport();
+        }
+    }
+
+Compile our Java code to generate our jar with all dependencies needed for.
+
+.. code-block:: shell
+
+    $ mvn clean install
+    $ cp target/cpptojava-1.0-SNAPSHOT-jar-with-dependencies.jar cpptojava.jar
+    $ cp cpptojava.jar <c++_project_path>
+
+**C++ Side**
+
+This process fetch Java JVM references to call methods needed for, via the C Data Interface:
+
+.. code-block:: cpp
+
+    #include <iostream>
+    #include <arrow/api.h>
+    #include <arrow/c/bridge.h>
+    #include <jni.h>
+
+    JNIEnv* create_vm(JavaVM ** jvm) {
+        JNIEnv *env;
+        JavaVMInitArgs vm_args;
+        JavaVMOption options[2];
+        options[0].optionString = "-Djava.class.path=cpptojava.jar"; // java jar name
+        options[1].optionString = "-DXcheck:jni:pedantic";
+        vm_args.version = JNI_VERSION_1_8;
+        vm_args.nOptions = 2;
+        vm_args.options = options;
+        int status = JNI_CreateJavaVM(jvm, (void**) &env, &vm_args);
+        if (status < 0) printf("\n<<<<< Unable to Launch JVM >>>>>\n");
+        return env;
+    }
+
+    int main() {
+        JNIEnv* env;
+        JavaVM* jvm;
+        env = create_vm(&jvm);
+        if (env == NULL) return 1;
+        jclass  javaClassToBeCalledByCpp = NULL;
+        javaClassToBeCalledByCpp = env ->FindClass("ToBeCalledByCpp");
```

Comment:
Can you make sure to format the source code?

Reply:
Reformat code added